services/catalog/src/simcore_service_catalog/core/background_tasks.py

```diff
@@ -0,0 +1,177 @@
+        possible_gid = await groups_repo.get_gid_from_affiliation(user_affiliation)
+        if possible_gid:
+            reader_gids.append(possible_gid)
+            if not owner_gid:
+                owner_gid = possible_gid
+
+    # if no owner gid yet, we pass readable rights to the everyone group
+    if not owner_gid:
+        reader_gids.append(everyone_gid)
+
+    # we add the owner with full rights, unless it's everyone
+    access_rights = [
+        ServiceAccessRightsAtDB(
+            key=service.key,
+            version=service.version,
+            gid=gid,
+            execute_access=True,
+            write_access=(gid == owner_gid),
+        )
+        for gid in set(reader_gids)
+    ]
+
+    return (owner_gid, access_rights)
+
+
+async def _create_services_in_db(
+    app: FastAPI,
+    service_keys: Set[Tuple[ServiceKey, ServiceVersion]],
+    services: Dict[Tuple[ServiceKey, ServiceVersion], ServiceDockerData],
+) -> None:
+    engine: Engine = app.state.engine
+    async with engine.acquire() as conn:
+        services_repo = ServicesRepository(conn)
+        for service_key, service_version in service_keys:
+            service: ServiceDockerData = services[(service_key, service_version)]
+            # find the service owner
+            owner_gid, service_access_rights = await _get_service_access_rights(
+                service, conn
+            )
+            # set the service in the DB
+            await services_repo.create_service(
+                ServiceMetaDataAtDB(**service.dict(), owner=owner_gid),
+                service_access_rights,
+            )
+
+
+async def sync_registry_task(app: FastAPI) -> None:
+    # get list of services from director
+    while True:
+        try:
+            logger.debug("syncing services between registry and database...")
+            services_in_registry: Dict[
+                Tuple[ServiceKey, ServiceVersion], ServiceDockerData
+            ] = await _list_registry_services(app)
+            services_in_db: Set[
+                Tuple[ServiceKey, ServiceVersion]
+            ] = await _list_db_services(app)
+
+            # check that the db has all the services at least once
+            missing_services_in_db = set(services_in_registry.keys()) - services_in_db
+            if missing_services_in_db:
+                logger.debug(
+                    "missing services in db:\n%s", pformat(missing_services_in_db),
+                )
+                # update db (rationale: missing services are shared with everyone for now)
+                await _create_services_in_db(
+                    app, missing_services_in_db, services_in_registry
+                )
+
+            await asyncio.sleep(app.state.settings.background_task_rest_time)
+
+        except CancelledError:
+            # task is stopped
+            return
+        except Exception:  # pylint: disable=broad-except
+            logger.exception("some error occured")
```

Comment:
```suggestion
        except Exception:  # pylint: disable=broad-except
            logger.exception("some error occured")
            await asyncio.sleep(5)
```

In case of error this will retry a lot of times in a row and should be limited. 5 seconds should be ok

Reply:
done